lib/src/rules/overridden_fields.dart

```diff
@@ -75,7 +75,11 @@ Iterable<InterfaceType> _findAllSupertypesAndMixins(
 
   accumulator.add(interface);
   InterfaceType superclass = interface.superclass;
-  Iterable<InterfaceType> interfaces = [superclass]
+  List<InterfaceType> interfaces = [];
+  if (superclass != null) {
+    interfaces.add(superclass);
```

Comment:
Hm... This whole function looks like a copy of `ClassElement.allSupertypes`. This method receives `InterfaceType`, but its client actually uses `classElement.type`, and actually does not care about types - just about presence of an accessor in a supertype.

Reply:
Nope. This excludes implemented types, looking only along the superclass chain.